content/index.md

```diff
@@ -0,0 +1,172 @@
+
+> As a .NET developer, I want to use the Nexmo CLI to create an application and buy a number so I can follow a Nexmo tutorial.
+
+> I have signed up for Nexmo but we use multiple providers. I want to look up prices for various operations so I can prepare price forecasts and decide which provider to use for which aprt of my application.
+
+> As a developer, I want to receive SMS messages in my application.
+
+> As a developer, I want to buy a number, and have my application respond to incoming calls on this number.
+
+> As a developer, I use ngrok for local development. I don't have a paid account, so I often need to reconfigure my webhook URL (sometimes only one of them).
+
+> As a developer, I am creating an application that sends and receives SMS, using the Messages API for outgoing messages.
+
+> As a developer, I need to inspect the configuration of my account and may also want to change one or more settings.
+
+> As a developer, I haven't used Nexmo since I first looked at it a few weeks ago. I want to look at which applications I have and which numbers are used by which application so that I can decide if I can reallocate or cancel numbers.
+
+> As a developer I want to look up some information about a user's number for security purposes.
+
+> I have been asked to detail for our auditors all the numbers on our Nexmo account and which applications use them. I need to supply this as a spreadsheet.
+
+> As a developer, we run a large number of applications, and each one may have multiple incoming numbers linked to it. I want to know which numbers are linked to application X.
+
+
+## Packaging, Installation and Setup
+
+CLI tools **must**:
+
+ - be available on all platforms (Windows, Mac, Linux - any others would be a bonus)
+ - have as few dependencies as possible
+ - be open to users of all our supported technology stacks (Ruby, PHP, Python, Java, NodeJS, .NET, Go, Android, iOS)
+
+In addition, it would be nice if the tool could:
+
+ - upgrade itself
+ - be extensible with plugins
+ - be relatively small in size
+ - not store user's actual credentials on disk
+
+The tool will be able to state its current version when `--version` is supplied
+
+## Command Design
+
+Consistency and predictability are the key to a command line tools that users become familiar and comfortable with. To this end, all commands will follow the same conventions to enable us to build and maintain a tool that continues to feel familiar.
+
+```
+awesomecli topic:command --kebab-case-param value 
+```
+
+The above shows a basic template to use as a starting point
+
+* **`topic`** will usually be a plural noun. Examples could include `applications`, `numbers`, `insights`.
+* **`command`** will usually be an active or imperative verb. Examples could include `create`, `list`, `configure`.
+* For more complex areas, add a **subtopic** with the format `topic:subtopic:command`. Examples could include `accounts:jwt:generate`, `applications:numbers:link`.
```

Comment:
I'm not 100% sure on subtopics - it feels like we're replacing a flag with a namespace e.g. `applications:numbers:link <id>` rather than `numbers:link --application <id>`

Reply:
I don't like the idea of subtopics either, we'd still need to add a parameter for the subtopic, we can't have variable subtopics AFAIK

Reply:
I think noun - verb might be a better way of structuring this? I also happen to prefer `number` over `numbers` for the `link` verb - unless it supports linking multiple numbers in a single execution!

Reply:
I think the example I picked wasn't very good either!  The linking/unlinking stuff really needs some thought about the developer experience rather than the API calls we need to make. It's the thing I look up every single time, and I think it's also changed by some of the upcoming API changes.

For singular/plural, I don't mind what we do but it would be excellent to pick one (and probably alias the other) and use it everywhere. Consistency is king!

Reply:
*Consistency is king!*

Reply:
Link number is a particularly interesting one, because we may be able to do it without specifying what type of thing they're linking to - we can infer it from the value, instead.